### ISSUE TITLE
Query argument in `retriever.get_chat_history()`

### ISSUE THREAD

**Issue by shreehari-aiplanet** [aiplanethub/genai-stack#105]
* GenAI Stack version: 0.2.5
* Python version: 3.8
* Operating System: Ubuntu

### Description

Query arguement in `retriever.get_chat_history()` seems a bit ambiguous. Hence it can be removed.

**Comment by dyaramatreya** [comment]
Hi @shreehari-aiplanet 
Do you mean to remove the query at https://github.com/aiplanethub/genai-stack/blob/f15b5b32aa7471535889d24845658ace98ccf614/genai_stack/retriever/base.py#L42?

or to remove at https://github.com/aiplanethub/genai-stack/blob/f15b5b32aa7471535889d24845658ace98ccf614/genai_stack/retriever/base.py#L38

**Comment by shreehari-aiplanet** [comment]
@dyaramatreya It needs to be removed wherever it is declared. Retreiver's chat history is basically getting the chat history from memory. A **mediator** between them is responsible for the interaction between **memory and retriever** to get the chat history. Hence it needs to be removed from more than 2 places.

**Comment by dyaramatreya** [comment]
@shreehari-aiplanet So basically, we have to delete this retriever.get_chat_history() method definition as well as the places where it is used. Am I correct?

**Comment by shreehari-aiplanet** [comment]
> delete this retriever.get_chat_history() method

No, `get_chat_history()` method should exist and the `query`  parameter that is being passed to it should be removed.

**Comment by dyaramatreya** [comment]
Ok @shreehari-aiplanet. Can I try working on this? Can you assign this issue to me? I'll make the changes and raise the PR

**Comment by dyaramatreya** [comment]
But then the query parameter passed at https://github.com/aiplanethub/genai-stack/blob/f15b5b32aa7471535889d24845658ace98ccf614/genai_stack/stack/mediator.py#L72 must also be removed right?

**Comment by shreehari-aiplanet** [comment]
> But then the query parameter passed at
> 
> https://github.com/aiplanethub/genai-stack/blob/f15b5b32aa7471535889d24845658ace98ccf614/genai_stack/stack/mediator.py#L72
> 
> must also be removed right?

yes

**Comment by dyaramatreya** [comment]
Thanks. Can you please assing this issue to me? I'll raise the PR in few hours

**Comment by dyaramatreya** [comment]
@shreehari-aiplanet Please have a look at the PR raised above and suggest if it's good or any changes are needed.
Thanks
